From 405382d953a097375d3db8dbb533370175302e21 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 9 Oct 2025 10:56:54 +1000
Subject: [PATCH] hwdb: don't tag a named Mouse device as pointingstick

The generic kernel hid drivers split up devices based on the application
collection, appending a suffix for each collection (e.g. Touchpad,
Mouse, ...). Many i2c touchpads get a "... Mouse" event node which is
mislabelled as pointingstick by the input_id builtin, see commit
3d7ac1c655ec40f3829543072494dcdfb92dbc6b.

Closes: https://github.com/systemd/systemd/issues/36677
(cherry picked from commit c4f072aaadedd9029bf0bef2036fdab8a4a3c180)

Resolves: RHEL-72702
---
 hwdb.d/60-input-id.hwdb | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hwdb.d/60-input-id.hwdb b/hwdb.d/60-input-id.hwdb
index edc166c847..d32bfedf59 100644
--- a/hwdb.d/60-input-id.hwdb
+++ b/hwdb.d/60-input-id.hwdb
@@ -58,6 +58,10 @@
 
 # Sort by brand, model
 
+# Any i2c device with a Mouse suffix in the name is not a pointing stick
+id-input:i2c:*:name:*Mouse:
+ ID_INPUT_POINTINGSTICK=0
+
 # Code Mercenaries Hard- und Software GmbH Virtual RC USB
 id-input:modalias:input:b0003v07C0p1125*
  ID_INPUT_MOUSE=
